src/stores/utils/contract.js

```diff
@@ -66,16 +69,41 @@ export const totalBurntCoins = async (contract) => {
   return new BN(burntCoins)
 }
 
-export const getBridgeValidators = async (bridgeValidatorContract) => {
-  let ValidatorAdded = await bridgeValidatorContract.getPastEvents('ValidatorAdded', {fromBlock: 0});
-  let ValidatorRemoved = await bridgeValidatorContract.getPastEvents('ValidatorRemoved', {fromBlock: 0});
-  let addedValidators = ValidatorAdded.map(val => {
-    return val.returnValues.validator
-  })
-  const removedValidators = ValidatorRemoved.map(val => {
-    return val.returnValues.validator
+export const getValidatorList = async (address, eth) => {
+  const v1Contract = new eth.Contract(BridgeValidatorsV1Abi, address);
+  const v1ValidatorsEvents = await getValidatorEvents(v1Contract)
+
+  const contract = new eth.Contract(BRIDGE_VALIDATORS_ABI, address);
+  const validatorsEvents = await getValidatorEvents(contract)
+
+  const rewardableContract = new eth.Contract(REWARDABLE_BRIDGE_VALIDATORS_ABI, address);
+  const rewardableValidatorsEvents = await getValidatorEvents(rewardableContract)
+
+  const eventList = [...v1ValidatorsEvents, ...validatorsEvents, ...rewardableValidatorsEvents]
+
+  const sortedEvents = eventList.sort((a, b) => a.blockNumber - b.blockNumber)
+
+  return processValidatorsEvents(sortedEvents)
+}
+
+const processValidatorsEvents = (events) => {
+  const validatorList = new Set()
+  events.forEach(event => {
+    if(event.event === 'ValidatorAdded') {
+      validatorList.add(event.returnValues.validator)
+    } else if(event.event === 'ValidatorRemoved') {
+      validatorList.delete(event.returnValues.validator)
+    }
   })
-  return addedValidators.filter(val => !removedValidators.includes(val));
+  return Array.from(validatorList)
+}
+
+const getValidatorEvents = async (bridgeValidatorContract) => {
+  try {
+    return await bridgeValidatorContract.getPastEvents('allEvents', { fromBlock: 0 })
```

Comment:
Looking the events starting from the block `0` could take lots of time. May we use `deployedAtBlock()` here?

Reply:
Good Idea! Updated to use `deployedAtBlock` when we get validators events and also bridge statistics https://github.com/poanetwork/bridge-ui/pull/184/commits/0f29ac3701d7fb08e894823330fac7032bcfb1d2